qiskit_nature/problems/second_quantization/base_problem.py

```diff
@@ -19,29 +19,53 @@
 
 from qiskit.opflow import PauliSumOp, Z2Symmetries
 
+from qiskit_nature import QiskitNatureError
 from qiskit_nature.drivers.second_quantization import BaseDriver, QMolecule, WatsonHamiltonian
 from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.properties.second_quantization.driver_result import DriverResult
 from qiskit_nature.results import EigenstateResult
+from qiskit_nature.transformers import BaseTransformer as LegacyBaseTransformer
 from qiskit_nature.transformers.second_quantization import BaseTransformer
 
 
 class BaseProblem(ABC):
     """Base Problem"""
 
-    def __init__(self, driver: BaseDriver, transformers: Optional[List[BaseTransformer]] = None):
+    def __init__(
+        self,
+        driver: BaseDriver,
+        transformers: Optional[List[Union[LegacyBaseTransformer, BaseTransformer]]] = None,
+    ):
         """
 
         Args:
             driver: A driver encoding the molecule information.
             transformers: A list of transformations to be applied to the molecule.
+
+        Raises:
+            QiskitNatureError: if new and legacy transformer instances are mixed.
         """
 
         self.driver = driver
         self.transformers = transformers or []
 
+        self._legacy_transform = False
+        new_transformers = False
+        for trafo in self.transformers:
+            if trafo is None:
+                continue
+            if isinstance(trafo, LegacyBaseTransformer):
+                self._legacy_transform = True
+            elif isinstance(trafo, BaseTransformer):
+                new_transformers = True
+        if self._legacy_transform and new_transformers:
+            raise QiskitNatureError("You cannot mix transformers of the legacy and the new type!")
```

Comment:
Nit picking really.... 'You cannot mix...' is not exactly correct since clearly they did! A more accurate message would be more like `A mix of current and deprecated transformers is not supported`